### ISSUE TITLE
[BUG] failed to launch

### ISSUE THREAD

**Issue by Brli** [Zettlr/Zettlr#306]
## Description
```
$ electron /usr/lib/zettlr/

[19499:0907/165331.024212:ERROR:buffer_manager.cc(488)] [.DisplayCompositor]GL ERROR :GL_INVALID_OPERATION : glBufferData: <- error from previous GL command
```
given a blank (white) window with menubar

if using the binary package of .rpm, it gives even less info, nothing shown under terminal, and a blank window is shown

## Reproducing
run from terminal under Arch Linux

## Expected behaviour
launched fine with a work space

**Screenshots**
![Screenshot_20190907_165548](https://user-images.githubusercontent.com/4042566/64472394-5ac70900-d190-11e9-99d0-5977b6f8d61c.png)

## Platform
 - OS: [e.g. iOS] Arch Linux
 - OS Version [e.g. 11]
 - Zettlr Version [e.g. 0.15] 1.4.0
 - Screen Resolution [only necessary if the bug is relating to the GUI]

## Additional information
Tried to clean the config files from `~/.config/zettlr`, doesn't help, it still failed to launch

**Comment by remyd1** [comment]
Hi,

I have the same error message on Fedora (30) with Gnome, when launching it from terminal (+ some other warnings). However, Zettlr is able to run, with the menu, but I cannot open anything in it.

```
Fontconfig warning: "/usr/share/fontconfig/conf.avail/05-reset-dirs-sample.conf", line 6: unknown element "reset-dirs"
[8275:0907/112823.024604:ERROR:buffer_manager.cc(488)] [.DisplayCompositor]GL ERROR :GL_INVALID_OPERATION : glBufferData: <- error from previous GL command
Error creating proxy: Transport « disabled » inconnu ou non pris en charge pour l’adresse « disabled: » (g-io-error-quark, 13)
Error creating proxy: Transport « disabled » inconnu ou non pris en charge pour l’adresse « disabled: » (g-io-error-quark, 13)
Error creating proxy: Transport « disabled » inconnu ou non pris en charge pour l’adresse « disabled: » (g-io-error-quark, 13)
Error creating proxy: Transport « disabled » inconnu ou non pris en charge pour l’adresse « disabled: » (g-io-error-quark, 13)
Error creating proxy: Transport « disabled » inconnu ou non pris en charge pour l’adresse « disabled: » (g-io-error-quark, 13)

(zettlr:8246): dconf-WARNING **: 11:28:44.007: failed to commit changes to dconf: Transport « disabled » inconnu ou non pris en charge pour l’adresse « disabled: »
```


My system:
Fedora release 30 (Thirty)

```
<?xml version="1.0"?>
<gnome-version>
 <platform>3</platform>
 <minor>32</minor>
 <micro>2</micro>
 <distributor>GNOME.Org</distributor>
 <date>2019-05-07</date>
</gnome-version>
```

**Comment by mseliger** [comment]
Same bug as before with Lubuntu (LxQT) 19.04
[10085:0907/132057.998524:ERROR:buffer_manager.cc(488)] [.DisplayCompositor]GL ERROR :GL_INVALID_OPERATION : glBufferData: <- error from previous GL command

(zettlr:10057): GLib-GIO-CRITICAL **: 13:21:01.087: g_dbus_proxy_new: assertion 'G_IS_DBUS_CONNECTION (connection)' failed

Zettlr started, but I can not open any folder - it just remains as there is no folder opened.

**Comment by nathanlesage** [comment]
FIREWORKS OF BUGS! 🐛

So, had a quick look around the net and here's what I've found so far:

Concerning the Lubuntu `glib`-error, it seems to be something with the Kernel: https://bugs.launchpad.net/ubuntu/+source/leafpad/+bug/1770129

Apparently the other `glib`-errors on Arch can be explained due to a `glibc`-update. So nothing Zettlr-related here. If you continue to encounter these issues, I'd suggest reporting the issue on github.com/electron/electron.

The `dconf`-error, here's a possible solution: https://unix.stackexchange.com/a/474830. I'm not sure where the rights-issue begins, but in case I'd recommend backing up config, stats and stuff and then removing the full directory. Chromium/Electron puts a lot of cached data there and in the past we've been able to solve many issues simply by removing the cache.

But now to the Zettlr stuff:

> Zettlr started, but I can not open any folder - it just remains as there is no folder opened.

Concerning this: Would you mind starting Zettlr from the command line and look for any messages? It seems to me there's an error somewhere along the line that's unfortunately only printed on the console …

**Comment by samarulmeu** [comment]
On Lubuntu 18 it works just fine, although it seems a little slower than the beta versions, but this I'm still investigating...

**Comment by Brli** [comment]
It's hard to tell, the other electron apps using the same electron binary Arch provided worked fine, eg. draw.io, riot-desktop.

They do have the same error message, but then, they outcome fine, display things.

But zettlr stays blank...nor the binary version which supposedly using the built-in electron binary

**Comment by Brli** [comment]
`*-beta` works, but start with `*-beta.1`, it is this situation...

wonders if it has something to do with vue?

**Comment by mseliger** [comment]
The last beta version 1.4. Beta 2 works fine for me. There is no problem when changing a folder. The installation size of version 1.4 has not the same size as 1.4 Beta 2. Is in 1.4 something missing?

**Comment by nathanlesage** [comment]
No, in 1.4 is nothing missing, I've switched to Electron 6, which was the problem. Please refer to this issue for further information on why the dialogs stopped working: https://github.com/Zettlr/Zettlr/issues/308

And it certainly has nothing to do with Vue. Basically everything in this thread (apart from the dialogs, for which there is above mentioned issue) pertains to issues Electron 6, so the framework, not Zettlr, has introduced into Linux.

**Comment by BAKFR** [comment]
About this:

> > Zettlr started, but I can not open any folder - it just remains as there is no folder opened.
> 
> Concerning this: Would you mind starting Zettlr from the command line and look for any messages? It seems to me there's an error somewhere along the line that's unfortunately only printed on the console …

The bug appears on an empty config only, when there is no open folder. I've looked everywhere, there is no error log. The "save file" silently fails (since there is no selected folder), and we can't open any folder: the selection dialogbox appears, but nothing happens after we select a folder.

**Comment by Brli** [comment]
for the record, this commit fix the issue for ver >= 1.4.0-b.2:
https://aur.archlinux.org/cgit/aur.git/commit/PKGBUILD?h=zettlr&id=cede21459bc858f6c7b70230d4b72fff41a47343

yes, it is a problem with Vue, that we must webpack it in order to make the components to render properly, or electron would just return a blank page.

**Comment by nathanlesage** [comment]
> yes, it is a problem with Vue, that we must webpack vue after the node/electron components are generated in order to make the vue components to render properly, or electron would just return a blank page.

Yes, it's an additional command that has to be run since the last version for a successful build: `yarn wp:prod` (or `wp:dev` for development/console output in the renderer)

Thank you for your efforts!
